src/engine/Collision/BoundingBox.ts

```diff
@@ -120,23 +120,46 @@ export class BoundingBox implements ICollidable {
       var tmin = -Infinity;
       var tmax = +Infinity;
 
-      if (ray.dir.x !== 0) {
-         var tx1 = (this.left - ray.pos.x) / ray.dir.x;
-         var tx2 = (this.right - ray.pos.x) / ray.dir.x;
-         tmin = Math.max(tmin, Math.min(tx1, tx2));
-         tmax = Math.min(tmax, Math.max(tx1, tx2));
-      }
-
-      if (ray.dir.y !== 0) {
-         var ty1 = (this.top - ray.pos.y) / ray.dir.y;
-         var ty2 = (this.bottom - ray.pos.y) / ray.dir.y;
-         tmin = Math.max(tmin, Math.min(ty1, ty2));
-         tmax = Math.min(tmax, Math.max(ty1, ty2));
-      }
+      var xinv = ray.dir.x === 0 ? 999999999999 : (1 / ray.dir.x);
```

Comment:
That might be the best thing to do, the 9's were a quick and dirty way to accomplilsh the same thing. I'll fix that.